### PR TITLE
Friendshipsv1

### PR DESCRIPTION
**Pull request by m15e**
- [ ] Created a model with associations and requested features for friendships

- [ ] Created tests

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -9,6 +9,8 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
-  has_many :requestees, through: :friends, source: 'user', dependent: :destroy
-  has_many :requestors, through: :friends, source: 'user', dependent: :destroy
+  has_many :friendships
+  has_many :requests, class_name: :friendship, foreign_key: 'friend_id'
+  scope :current_user_friends, -> { where('user_id == ?', current_user.id) }
+  # scope :not_user_friends, -> { where('user_id != ?', current_user.id) }
```
Comment (BrittanyBlake) [inline app/models/user.rb]: - [ ] Please remove commented code. It is bad practice to leave it in.

**Review by BrittanyBlake — CHANGES_REQUESTED** [review]
### Status: Changes Required ♻️ 
****
### Hi @defoebrand  and @m15e ! 👋 
 You have also done a wonderful job with your project so far! 🤩 🎖️ It looks really good! 👏 👏 👏  There are just a few issues that need to be resolved before you can move on: ⬇️ 
****

- [ ] The User's timeline should only show posts from the logged in users and their friends. 
screenshot of the requirements 👇 
Screen Shot 2020-09-21 at 7 54 16 PM
screenshot from your webpage 👇 
Screen Shot 2020-09-21 at 7 54 44 PM
Screen Shot 2020-09-21 at 7 54 54 PM
   - As you can see, the user is not friends with Brittany but her post appears on the timeline. 
****
### Please make the required changes and resubmit for a new code review once you are ready 🙏

Good Luck and Happy Coding 🍀 👩🏾‍💻

_P.S. If you have any questions about my review or need help, you can message me on slack (@ Brittany Blake)._

**Review by andresporras3423 — CHANGES_REQUESTED** [review]
### Status: Changes Required :recycle:
**Hello Mark Rode & Defoe Defoe** 
Great job following the improvements of the last review :muscle:  :muscle: 
Your project meets most of the requirements :clap: :clap: :clap:  
We still need to fix some details:

- [x] After creating a post it should immediately available in "timeline" and "stay in touch".
- [x] After I reject a friendship request I should be able to request friendship to that same user.
- [x] After my friendship request is rejected I should be able to request a friendship again to that same user.
- [x] include tests for negative scenarios, you might use "to_not eql" and "to_not be_an".

Please request another review once you are ready! :+1: 

Happy coding :relaxed:

Oscar Andrés Russi Porras

**Comment by defoebrand** [comment]
All reviewer requests have been implemented except the last one. The model tests currently in place include numerous to eq false, which is the same as saying to not eql true. Unless there is a need for specifically calling out 'to not eql' then we have already implemented many negative test scenarios.

**Review by uche-inyama — APPROVED** [review]
## Approved

### Hi team :wave: 

Go ahead with the next milestone.

#### ``` Have a great day ``` :slightly_smiling_face: 

``` Uchechukwu Henry Inyama ```
